### PR TITLE
Prepublish Changes - Updated tooling and build test

### PR DESCRIPTION
**Pull request by xavierchanth**
This is a stacked PR with 5 total branches, all of which involve changes to prepare at_app, at_app_create & at_app_bundler for publishing. It should be reviewed in order of 1,2,3,4,5, but merged in order of 5,4,3,2,1.

Branches:
- **xavierchanth-4.2.2-prepublish-1**
- xavierchanth-4.2.2-prepublish-2 #97 
- xavierchanth-4.2.2-prepublish-3 #98
- xavierchanth-4.2.2-prepublish-4 #99
- xavierchanth-4.2.2-prepublish-5 #100

More information on stacked PRs [here](https://www.michaelagreiler.com/stacked-pull-requests/)

**- What I did**

- Added setup scripts
  - setup:all which does all necessary repo setup (outside of setting up melos)
  - setup:mason - which installs the correct version of mason_cli
    (since there are commonly breaking changes due to it being in development)

- Renamed the appropriate scripts from build to bundle:
  - build:bundler -> bundle:bundler (yes, this is very meta, I have a bundle script for my bundler)
  - build:create -> bundle:create
  - build:templates -> bundle:templates
  - Renamed the same script filenames as well

- Updated the script names in the build test

- Added a --no-analyze flag to all bundle scripts
  - Enabled it for all bundle commands in the build test
  - Separately called analyze for each in the build test
 
**- How I did it**

Local changes

**- How to verify it**

Review and merge all 5 PRs in the stack.
Then verify with the build_test.

**- Description for the changelog**
Prepublish Changes - Updated tooling and build test

### CODE REVIEW

**Comment by gkc** [comment]
@XavierChanth checked out the branch, problem running dart pub get / pub upgrade in at_app:
```
gkc2019:at_app gary$ dart pub upgrade
Resolving dependencies... (1.8s)
Because at_app depends on at_app_create any which doesn't exist (could not find package at_app_create at https://pub.dartlang.org), version solving
  failed.
```

**Comment by xavierchanth** [comment]
> @XavierChanth checked out the branch, problem running dart pub get / pub upgrade in at_app:
> 
> ```
> gkc2019:at_app gary$ dart pub upgrade
> Resolving dependencies... (1.8s)
> Because at_app depends on at_app_create any which doesn't exist (could not find package at_app_create at https://pub.dartlang.org), version solving
>   failed.
> ```

Please install melos:
```
dart pub global activate melos
```

And use it to link the packages:
```
melos bootstrap
```

at_app_create hasn't been published yet, hence why a dart pub get won't resolve it.

**Comment by gkc** [comment]
Ah - melos bootstrap was the thing I was missing; I had already figured out
`dart pub global activate melos`

File: tools/setup_all.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+if [ "$GITHUB_ACTION" ];
```
Comment (cpswan) [inline tools/setup_all.sh]: needs ! -z

**Review by cpswan — APPROVED** [review]

**Pull request merged**